### PR TITLE
feat(ST-140): add teacher profile page component

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added User Profile module
- Added Teacher Profile component
- Added rtk query to fetch profile info
- Added a user profile container 
- Showed teacher profile data

## Associated Ticket Link(s)

- https://trello.com/c/uNRGjm4P

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings
Screen Shot 2024-09-28 at 11 30 06 AM

### CODE REVIEW

File: modules/UserProfile/components/TeacherProfileInfo/TeacherProfileInfo.styles.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    color: theme.colors.green[7],
+    textAlign: "center",
+    // padding: "8px",
+  },
+
+  tableData: {
+    fontWeight: "bold",
+    color: theme.colors.green[7],
+    textAlign: "center",
+    // padding: "8px",
```
Comment (ishmam-iqbal-sazim) [inline modules/UserProfile/components/TeacherProfileInfo/TeacherProfileInfo.styles.ts:38]: Same comment as above

File: modules/UserProfile/components/TeacherProfileInfo/TeacherProfileInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            Subjects To Teach
+          </Text>
+          <Box className={classes.subjectsList}>
+            <Table withColumnBorders>
+              <tbody>
+                <tr>
+                  <td className={classes.tableSL}>SL</td>
+                  <td className={classes.tableData}>Subject</td>
+                </tr>
+                {subjects?.map((subject, idx) => (
```
Comment (ishmam-iqbal-sazim) [inline modules/UserProfile/components/TeacherProfileInfo/TeacherProfileInfo.tsx:43]: Should do a subjects.length check here as well

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Review by ragib-hossain-sazim — APPROVED** [review]

File: modules/UserProfile/components/TeacherProfileInfo/TeacherProfileInfo.tsx (hunk 2 above)
Comment (faysal-alam-sazim) [reply modules/UserProfile/components/TeacherProfileInfo/TeacherProfileInfo.tsx:43]: Bhai I am showing a loading components when data is loading. This is a required field length can't be zero. Why should I check length then?

**Pull request merged**